macros/src/lib.rs

```diff
@@ -22,6 +22,16 @@ extern crate mac;
 
 use rustc::plugin::Registry;
 
+macro_rules! panictry {
+    ($e:expr) => ({
+        use syntax::diagnostic::FatalError;
+        match $e {
+            Ok(e) => e,
+            Err(FatalError) => panic!(FatalError)
+        }
+    })
+}
+
```

Comment:
This is what libsyntax does:

https://github.com/rust-lang/rust/blob/b9ed9e2a326cafa6f2445aec7221dc419089cf5f/src/libsyntax/lib.rs#L53-L56
> A variant of 'try!' that panics on Err(FatalError). This is used as a crutch on the way towards a non-panic!-prone parser. It should be used for fatal parsing errors; eventually we plan to convert all code using panictry to just use normal try

Right now it’s the same as `.unwrap`, but the plan is apparently that it should be `try!` eventually.